tests/auth_plugin_test.php

```diff
@@ -843,6 +844,84 @@ public function test_ipmismatch_exception_thrown_if_ip_is_incorrect() {
     }
 
     /**
+     * Test that IP address mismatch exception gets thrown if incorrect IP.
+     *
+     * @expectedException moodle_exception
+     * @expectedExceptionMessage Unsupported redirect to http://www.example.com/moodle detected, execution terminated.
+     */
+    public function test_ipmismatch_exception_notthrown_if_ip_is_whitelisted() {
+        global $DB;
+
+        set_config('ipwhitelist', '10.0.0.0/8;172.16.0.0/12;192.168.0.0/16', 'auth_userkey');
+
+        $key = new stdClass();
+        $key->value = 'IpmismatchKey';
+        $key->script = 'auth/userkey';
+        $key->userid = $this->user->id;
+        $key->instance = $this->user->id;
+        $key->iprestriction = '192.168.1.1';
+        $key->validuntil    = time() + 300;
+        $key->timecreated   = time();
+        $DB->insert_record('user_private_key', $key);
+
+        $_POST['key'] = 'IpmismatchKey';
+        $_SERVER['HTTP_CLIENT_IP'] = '192.168.1.2';
+        @$this->auth->user_login_userkey();
+    }
+
+    /**
+     * Test that IP address mismatch exception gets thrown if incorrect IP.
```

Comment:
Is it correct test description?

Reply:
No, was copy and pasted and not changed, fixed.